### DISCUSSION TITLE
Training in french language is it possible ?

### DISCUSSION THREAD

**Discussion by WillIsback** [modelscope/FunASR#1364] [Q&A] [answered]
I am french, and for learning and fun purpose I am trying to make a streaming instant translate program that : 
-get audio from mic / transcrib it / translate it / speech it with a trained tts voice model. 

FunASR looks promising for my project but it seems to be only english/chinese friendly. Do you have any will to add french or can I help somehow by training a model ?

Best regards,
William

**Comment by diego1mfbt** [comment]
Here are some good audio samples to practice french: https://afrancesados.com/frases-em-frances/

**Comment by LauraGPT** [comment, marked as answer]
Yes - for **French ASR today, you can start with an existing model instead of training one from scratch**. The answer is different from when this question was opened in 2024.

Current FunASR can run [Qwen3-ASR-0.6B / 1.7B](https://huggingface.co/Qwen/Qwen3-ASR-0.6B), whose published language table explicitly includes French (`fr`) and supports both offline and streaming inference. A minimal offline example is:

```bash
pip install -U funasr "qwen-asr==0.0.6" "transformers==4.57.6" accelerate
```

```python
from funasr import AutoModel

model = AutoModel(
    model="Qwen/Qwen3-ASR-0.6B",
    hub="hf",
    device="cuda:0",
    dtype="bf16",
)

result = model.generate(input="french.wav", language="fr")
print(result[0]["text"])

# Omit language="fr" if you want automatic language detection.
```

This follows FunASR's current [Qwen3-ASR example](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/examples/industrial_data_pretraining/qwen3_asr/demo.py).

### Reproduced French check

I verified this with FunASR `1.3.14`, `qwen-asr==0.0.6`, and the public [`PolyAI/minds14`](https://huggingface.co/datasets/PolyAI/minds14) `fr-FR` sample `train[1]`:

```text
Reference:    Bonjour je voudrais changer d'adresse
language=fr: Bonjour, je voudrais changer d'adresse.
auto detect: Bonjour, je voudrais changer d'adresse.  (French)
```

The model was loaded from a pinned local Qwen3-ASR-0.6B snapshot on an H100, and all IPv4/IPv6 connections were blocked after imports. Both runs completed with **zero network attempts**, so the result was not produced by a remote API.

### Streaming microphone input

The current repository also contains a native incremental WebSocket server built on `qwen-asr[vllm]`:

```bash
pip install -U "qwen-asr[vllm]==0.0.6" "transformers==4.57.6" websockets numpy

cd examples/industrial_data_pretraining/qwen3_asr
python serve_qwen3_asr_ws.py \
  --model Qwen/Qwen3-ASR-0.6B \
  --port 10095 \
  --gpu-memory-utilization 0.8
```

The client sends 16 kHz mono little-endian `int16` PCM chunks and receives partial text. It sends the text message `STOP` to finalize the turn. See the [server source](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/examples/industrial_data_pretraining/qwen3_asr/serve_qwen3_asr_ws.py) and its [streaming/VAD notes](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/examples/industrial_data_pretraining/qwen3_asr/serve_qwen3_asr_ws_notes_en.md).

One important product boundary: incremental ASR does not decide automatically that the speaker's turn has ended. The example uses explicit `STOP`; for hands-free microphone use, add an endpoint/turn-detection VAD outside the ASR service.

Your complete application would therefore be:

```text
microphone -> French ASR -> translation model/service -> TTS voice
```

Qwen3-ASR handles transcription, not translation or TTS, so those remain separate stages.

### About training or fine-tuning

The current FunASR Qwen3-ASR integration is **inference-only**: its training `forward()` explicitly raises `NotImplementedError` ([source](https://github.com/modelscope/FunASR/blob/1d8080af99fad370645eec7de9d1bcc5ae4ab587/funasr/models/qwen3_asr/model.py#L193-L199)). The repository's existing [fine-tuning recipe](https://github.com/FunAudioLLM/Fun-ASR/blob/84c2c9dcd64f38daf6f560cc13e23a0251d4c9fa/docs/finetune.md) targets `FunAudioLLM/Fun-ASR-Nano-2512`; it is not a supported Qwen3-ASR French fine-tuning path. The published Fun-ASR-MLT-Nano language list also does not currently include French ([model table](https://github.com/FunAudioLLM/Fun-ASR/blob/84c2c9dcd64f38daf6f560cc13e23a0251d4c9fa/README.md#L32-L35)).

The most useful contribution first would be a reproducible French evaluation: test Qwen3-ASR on your microphone/domain, then report failing samples (with redistribution permission), expected transcripts, model version, and whether the error occurs offline or in streaming mode. That gives us actionable evidence for French quality improvements without asking you to build a new model unnecessarily.
